### PR TITLE
DOC-601: advcode search and replace

### PR DESCRIPTION
**Pull request by tylerkelly13**
Related Ticket: DOC-601

Description of Changes:
* Document new `advcode` search and replace functionality.
* Add `advcode` demo to plugin and enterprise page.
* Add release note for TinyMCE 5.4 release.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: _includes/misc/advcode-shortcuts.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Advanced Code Editor search and replace keyboard shortcuts
+
+Advanced Code Editor provides the following shortcuts for search and replace within the code dialog.
+
+| Action | PC | Mac |
+| ---- | ----- | ----- |
+| Find | Ctrl+F | Cmd+F |
+| Find next instance | Ctrl+G | Cmd+G |
+| Find previous instance | Shift+Ctrl+G | Shift+Cmd+G |
+| Replace | Ctrl+H | Cmd+H |
```
Comment (lnewson) [inline _includes/misc/advcode-shortcuts.md:10]: I thought we said we couldn't document `Cmd+H` on Mac as it "hides" the window in Safari?

Comment (tylerkelly13) [reply _includes/misc/advcode-shortcuts.md:10]: @lnewson @trevorjay did we remap it to something else?

File: _includes/misc/advcode-shortcuts.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+## Advanced Code Editor search and replace keyboard shortcuts
+
+Advanced Code Editor provides the following shortcuts for search and replace within the code dialog.
+
+| Action | PC | Mac |
```
Comment (lnewson) [inline _includes/misc/advcode-shortcuts.md:5]: Should `Cmd` be `Command` here? as that's how it's documented on the keyboard shortcuts page.

Comment (tylerkelly13) [reply _includes/misc/advcode-shortcuts.md:5]: :facepalm:

File: _includes/misc/advcode-shortcuts.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+| ---- | ----- | ----- |
+| Find | Ctrl+F | Cmd+F |
+| Find next instance | Ctrl+G | Cmd+G |
+| Find previous instance | Shift+Ctrl+G | Shift+Cmd+G |
+| Replace | Ctrl+H | Cmd+H |
+| Replace All | Shift+Ctrl+R | Shift+Cmd+Option+F |
+
+For a list of shortcuts available in Advanced Code Editor, see: [CodeMirror - Sublime Text bindings demo](https://codemirror.net/demo/sublime.html).
```
Comment (lnewson) [inline _includes/misc/advcode-shortcuts.md:13]: Not sure we should be linking this... as nowhere do we say it's just a wrapped CodeMirror instance and that's normally kept hidden?

Comment (tylerkelly13) [reply _includes/misc/advcode-shortcuts.md:13]: yeah, wasn't sure either. I'll remove that.

File: _includes/codepens/advcode/index.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      <tr>
+          <th style="width: 50%"><h2>The Code Plugin</h2></th>
+          <th style="width: 50%"><h2>The Advanced Code Editor Plugin</h2></th>
+      </tr>
+  </thead>
+  <tbody>
+      <tr>
+          <td style="width: 50%">
+            <textarea class="codedemo">
+              <p>The code (<code>code</code>) plugin provides a dialog for viewing and editing the HTML of the editor content.</p>
```
Comment (lnewson) [inline _includes/codepens/advcode/index.html:12]: Should the first "code" be `Code` here, as it's the name of the plugin?

File: _includes/misc/advcode-shortcuts.md (hunk 1 above)
Comment (lnewson) [reply _includes/misc/advcode-shortcuts.md:10]: It still has the original shortcut mentioned on the demo page, ie: `Shift+Cmd+Option+F`. Not the nicest though, but that's all I believe works on Safari (Chrome/Firefox aren't impacted). @trevorjay can confirm.

Comment (trevorjay) [reply _includes/misc/advcode-shortcuts.md:10]: I can confirm that the hideous `Shift+Cmd+Opt+F` does work on Safari.

Comment (tylerkelly13) [reply _includes/misc/advcode-shortcuts.md:10]: Wait, isn't that replaceAll? @lnewson @trevorjay

Comment (lnewson) [reply _includes/misc/advcode-shortcuts.md:10]: Right, I copied the wrong shortcut sorry... should have been `Cmd+Option+F`

Comment (trevorjay) [reply _includes/misc/advcode-shortcuts.md:10]: `Cmd-Opt-F` is Replace and `Shift-Cmd-Opt-F` is replace all.

File: _includes/misc/advcode-shortcuts.md (hunk 2 above)
Comment (trevorjay) [reply _includes/misc/advcode-shortcuts.md:5]: `Cmd` is more of the standard within Apple circles but if it's consistently `Command` elsewhere in our documentation then that's cool.

File: _includes/codepens/advcode/index.html (hunk 4 above)
Comment (tylerkelly13) [reply _includes/codepens/advcode/index.html]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1594/commits/42b7fcf5c90819e237ab7dccd63ff44883971ee0

File: _includes/misc/advcode-shortcuts.md (hunk 3 above)
Comment (tylerkelly13) [reply _includes/misc/advcode-shortcuts.md]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1594/commits/42b7fcf5c90819e237ab7dccd63ff44883971ee0

File: _includes/misc/advcode-shortcuts.md (hunk 2 above)
Comment (tylerkelly13) [reply _includes/misc/advcode-shortcuts.md:5]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1594/commits/42b7fcf5c90819e237ab7dccd63ff44883971ee0

File: _includes/misc/advcode-shortcuts.md (hunk 1 above)
Comment (tylerkelly13) [reply _includes/misc/advcode-shortcuts.md]: Fixed in https://github.com/tinymce/tinymce-docs/pull/1594/commits/42b7fcf5c90819e237ab7dccd63ff44883971ee0

**Pull request merged**